apps/dashboard/src/components/data-table/transaction-table.tsx

```diff
@@ -1,128 +1,285 @@
-    cell: ({ row }) => <TextCell size={80}>{formatPrice(row.original.price)}</TextCell>,
-    enableSorting: false,
-  },
-  {
-    accessorKey: 'quantity',
-    header: ({ column }) => <DataTableColumnHeader column={column} columnTitle="Quantity" />,
-    cell: ({ row }) => <TextCell>{row.original.quantity}</TextCell>,
-    enableSorting: false,
-  },
-  {
-    accessorKey: 'test_mode',
-    header: ({ column }) => <DataTableColumnHeader column={column} columnTitle="Test Mode" />,
-    cell: ({ row }) => <div>{row.original.test_mode ? '✓' : ''}</div>,
-    enableSorting: false,
-  },
-  {
-    accessorKey: 'status',
-    header: ({ column }) => <DataTableColumnHeader column={column} columnTitle="Status" />,
-    cell: ({ row }) => <TextCell>{row.original.status ?? '—'}</TextCell>,
-    enableSorting: false,
-  },
-  {
-    accessorKey: 'created_at_millis',
-    header: ({ column }) => <DataTableColumnHeader column={column} columnTitle="Created" className="justify-end" />,
-    cell: ({ row }) => (
-      <div className="min-w-[120px] w-full text-right pr-2">{new Date(row.original.created_at_millis).toLocaleString()}</div>
-    ),
-    enableSorting: false,
-  },
-];
+function UserAvatarCell({ userId }: { userId: string }) {
+  const app = useAdminApp();
+  const user = app.useUser(userId);
 
-export function TransactionTable() {
+  if (!user) {
+    return <AvatarCell fallback='?' />;
+  }
+  return (
+    <Link href={`/projects/${encodeURIComponent(app.projectId)}/users/${encodeURIComponent(userId)}`}>
+      <div className="flex items-center gap-2 max-w-40 truncate">
+        <AvatarCell
+          src={user.profileImageUrl ?? undefined}
+          fallback={user.displayName?.charAt(0) ?? user.primaryEmail?.charAt(0) ?? '?'}
+        />
+        {user.displayName ?? user.primaryEmail}
+      </div>
+    </Link>
+  );
+}
+
+function TeamAvatarCell({ teamId }: { teamId: string }) {
   const app = useAdminApp();
-  const [filters, setFilters] = React.useState<{ cursor?: string, limit?: number, type?: 'subscription' | 'one_time' | 'item_quantity_change', customerType?: 'user' | 'team' | 'custom' }>({
-    limit: 10,
-  });
+  const team = app.useTeam(teamId);
+  if (!team) {
+    return <AvatarCell fallback='?' />;
+  }
+  return (
+    <Link href={`/projects/${encodeURIComponent(app.projectId)}/teams/${encodeURIComponent(teamId)}`}>
+      <div className="flex items-center gap-2 max-w-40 truncate">
+        <AvatarCell
+          src={team.profileImageUrl ?? undefined}
+          fallback={team.displayName.charAt(0)}
+        />
+        {team.displayName}
+      </div>
+    </Link>
+  );
+}
 
+function pickChargedAmountDisplay(entry: MoneyTransferEntry | undefined): string {
+  if (!entry) return '—';
+  const chargedAmount = entry.charged_amount as Record<string, string | undefined>;
+  if ("USD" in chargedAmount) {
+    return `$${chargedAmount.USD}`;
+  }
+  // TODO: Handle other currencies
+  return '—';
```

Comment:
can we show something else here just to make it clear that there are other currencies than USD? or throw an error or something like that?

Reply:
(we will probably add new currencies rather sooner than later, so this will make our life easier once we do)